### PR TITLE
Scripts/Commands: Convert remaining debug commands

### PR DESCRIPTION
**Pull request by Carbenium**
**Changes proposed:**

-  Core/Misc: Make QuestShareMessages enum smart and make Player::SendPushToPartyResponse typesafe
-  Scripts/Commands: Drop debug setvid command since the used API does not exist
-  Scripts/Commands: Convert remaining debug commands

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

### CODE REVIEW

File: src/server/game/Handlers/QuestHandler.cpp [hunk 1]
```diff
@@ -637,7 +637,7 @@ void WorldSession::HandleQuestPushResult(WorldPacket& recvPacket)
     {
         Player* player = ObjectAccessor::FindPlayer(guid);
         if (player)
-            player->SendPushToPartyResponse(_player, msg);
+            player->SendPushToPartyResponse(_player, QuestShareMessages(msg));
```
Comment (Treeston) [inline src/server/game/Handlers/QuestHandler.cpp:640]: `static_cast`

File: src/server/scripts/Commands/cs_debug.cpp [hunk 2]
```diff
@@ -226,65 +226,36 @@ class debug_commandscript : public CommandScript
...
-
-        char* fail1 = strtok(nullptr, " ");
-        uint8 failArg1 = fail1 ? (uint8)atoi(fail1) : 0;
-
-        char* fail2 = strtok(nullptr, " ");
-        uint8 failArg2 = fail2 ? (uint8)atoi(fail2) : 0;
-
         WorldPacket data(SMSG_CAST_FAILED, 5);
         data << uint8(0);
         data << uint32(133);
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:233]: any idea if this magic number is from any enum or other constant?

File: src/server/scripts/Commands/cs_debug.cpp [hunk 3]
```diff
@@ -226,65 +226,36 @@ class debug_commandscript : public CommandScript
...
 
-    static bool HandleDebugSendSellErrorCommand(ChatHandler* handler, char const* args)
+    static bool HandleDebugSendSellErrorCommand(ChatHandler* handler, SellResult error)
     {
-        if (!*args)
-            return false;
-
-        SellResult msg = SellResult(atoi(args));
-        handler->GetSession()->GetPlayer()->SendSellError(msg, nullptr, ObjectGuid::Empty, 0);
+        handler->GetSession()->GetPlayer()->SendSellError(error, nullptr, ObjectGuid::Empty, 0);
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:252]: We have `handler->GetPlayer()` now btw.

File: src/server/scripts/Commands/cs_debug.cpp [hunk 4]
```diff
@@ -481,39 +452,31 @@ class debug_commandscript : public CommandScript
...
-        if (!*args)
-            return false;
-
         char const* name = "test";
-        uint8 code = atoi(args);
 
         WorldPacket data(SMSG_CHANNEL_NOTIFY, (1+10));
-        data << code;                                           // notify type
-        data << name;                                           // channel name
+        data << uint8(type);
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:460]: `static_cast` (here and everywhere)

File: src/server/scripts/Commands/cs_debug.cpp [hunk 5]
```diff
@@ -481,39 +452,31 @@ class debug_commandscript : public CommandScript
...
+        data << name;
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:461]: just put the literal inline

File: src/server/scripts/Commands/cs_debug.cpp [hunk 6]
```diff
@@ -481,39 +452,31 @@ class debug_commandscript : public CommandScript
...
     {
-        if (!*args)
-            return false;
-
         char const* msg = "testtest";
-        uint8 type = atoi(args);
+
         WorldPacket data;
-        ChatHandler::BuildChatPacket(data, ChatMsg(type), LANG_UNIVERSAL, handler->GetSession()->GetPlayer(), handler->GetSession()->GetPlayer(), msg, 0, "chan");
+        ChatHandler::BuildChatPacket(data, type, LANG_UNIVERSAL, handler->GetSession()->GetPlayer(), handler->GetSession()->GetPlayer(), msg, 0, "chan");
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:473]: same here, put all of these literals inline

File: src/server/scripts/Commands/cs_debug.cpp [hunk 7]
```diff
@@ -1597,18 +1523,18 @@ class debug_commandscript : public CommandScript
...
+    static bool HandleDebugInstanceSpawns(ChatHandler* handler, Variant<uint32, ExactSequence<'e', 'x', 'p', 'l', 'a', 'i', 'n'>> optArg)
     {
         Player const* const player = handler->GetSession()->GetPlayer();
         if (!player)
             return false;
 
         bool explain = false;
         uint32 groupID = 0;
-        if (!stricmp(args, "explain"))
+        if (optArg.holds_alternative<ExactSequence<'e', 'x', 'p', 'l', 'a', 'i', 'n'>>())
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_debug.cpp:1534]: for the record, you can just check uint32 first (or use `.index`) 😛

File: src/server/shared/SharedDefines.h [hunk 8]
```diff
@@ -969,9 +969,10 @@ enum SpellEffects
     TOTAL_SPELL_EFFECTS                             = 165
 };
 
-enum SpellCastResult
+// EnumUtils: DESCRIBE THIS
+enum SpellCastResult : uint8
 {
-    SPELL_FAILED_SUCCESS = 0,
+    SPELL_FAILED_SUCCESS = 0, // SKIP
```
Comment (Treeston) [inline src/server/shared/SharedDefines.h:975]: any reason to skip this?

File: src/server/shared/SharedDefines.h [hunk 9]
```diff
@@ -3146,9 +3148,10 @@ enum WeatherType
 
 #define MAX_WEATHER_TYPE 4
 
-enum ChatMsg
+// EnumUtils: DESCRIBE THIS
+enum ChatMsg : uint8
 {
-    CHAT_MSG_ADDON                  = 0xFFFFFFFF, // -1
+    CHAT_MSG_ADDON                  = 0xFF, // -1
```
Comment (Treeston) [inline src/server/shared/SharedDefines.h:3154]: are you sure this is used as uint8 the change is and a no-op?

File: src/server/scripts/Commands/cs_debug.cpp (hunk 2 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_debug.cpp:233]: That's a spell id.
https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=133

Comment (Treeston) [reply src/server/scripts/Commands/cs_debug.cpp:233]: alright, then just add a comment or something

File: src/server/scripts/Commands/cs_debug.cpp (hunk 4 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_debug.cpp:460]: Hmm, that's the style we use in the packet classes, so I would leave this as is.

File: src/server/shared/SharedDefines.h (hunk 9 above)
Comment (Carbenium) [reply src/server/shared/SharedDefines.h:3154]: In the end it is sent as uint8.

https://github.com/TrinityCore/TrinityCore/blob/f3b7f9ed5cb8fdc8142b95bfd16bdfb4a7b10e26/src/server/game/Chat/Chat.cpp#L523

File: src/server/scripts/Commands/cs_debug.cpp (hunk 7 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_debug.cpp:1534]: Isn't it nice though? :stuck_out_tongue_closed_eyes:

File: src/server/shared/SharedDefines.h (hunk 8 above)
Comment (Carbenium) [reply src/server/shared/SharedDefines.h:975]: Nah, probably not. It's just not used anywhere but the client doesn't mind the value, so we can include it again.

File: src/server/game/Handlers/QuestHandler.cpp (hunk 1 above)
Comment (Carbenium) [reply src/server/game/Handlers/QuestHandler.cpp:640]: :+1:

File: src/server/scripts/Commands/cs_debug.cpp (hunk 3 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_debug.cpp:252]: Right :+1:

**Review by Treeston — APPROVED** [review]

**Pull request merged**